### PR TITLE
Challenge/use class styles for link component

### PR DESCRIPTION
**Pull request by keyy123**
# Description

[comment]: # 'Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.'

Apply styling for the Link component via class names. It also includes comments that leave warnings about inline-theme style override using the MUI sx prop and references to theme styling in code.  

Fixes # 830

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [X] Bug fix (non-breaking change which fixes an issue)

## Screenshots

[comment]: # 'Please include screenshots of your changes if relevant.'

|       Before        |       After        |
| :-----------------: | :----------------: |
| "screenshot before" | "screenshot after" |
|                                                             |

[Before and After](https://www.loom.com/share/4a2fa769ccd44e9d8c12ffe959fd8748 )

# How Has This Been Tested?

- [X] Cypress integration
- [X] Cypress component tests
- [X] Jest unit tests

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by keyy123** [comment]
My only question is should I remove the sx attribute from the `<box>` component?

File: src/components/Home.js [hunk 1]
```diff
@@ -130,13 +131,18 @@ export default function Home(props) {
...
+                style within Link Component so be careful */}
+
               <Box
-                sx={{
-                  color: 'white',
-                  '.MuiBox-root&:hover': {
-                    color: '#adadad',
-                  },
-                }}
+                sx={
```
Comment (RubenSmn) [inline src/components/Home.js:138]: I think these are fine so no need to change them here. But you did help me find a problem!

**Review by RubenSmn — CHANGES_REQUESTED** [review]
I left some notes. The key problem here is that we set an *inline* style `display: 'block'` we need to find a solution for that problem.

**Comment by RubenSmn** [comment]
@dadiorchen I just tried to add the style to the class like so
```js
  link: {
    display: 'block',
    color: theme.palette.text.primary,
    '& :hover': {
      color: theme.palette.text.primary,
    },
  },
```
And build it locally and it seems to work fine. Could you point out were this is happening?
All the `a` tags had this `class`
```js
.jss1 {
  color: #eee;
  display: block;
}
```

File: src/components/Link.js [hunk 2]
```diff
@@ -9,8 +9,10 @@ import React from 'react';
 // generate material-ui style hook
 const useStyles = makeStyles((theme) => ({
   link: {
-    color: theme.palette.text.primary,
+    color: theme.palette.text.primaryReverse,
```
[the remark this answers is not in the document]
Comment (keyy123) [reply src/components/Link.js:12]: Should I revert style to primary?

File: src/components/playground/FontAddInput.cy.js [hunk 3]
```diff
@@ -32,7 +32,12 @@ describe('Toggle Theme Mode', () => {
       </PlaygroundProvider>,
     );
     cy.get('.MuiAccordionSummary-content').click();
-    cy.get('.MuiFormControl-root').eq(0).type('void void vpid{enter}');
-    cy.get('.MuiFormControl-root').eq(0).contains('Font could not be loaded');
+    cy.get('.MuiFormControl-root').as('formControl');
```
[the remark this answers is not in the document]
Comment (keyy123) [reply src/components/playground/FontAddInput.cy.js:35]: Yes @RubenSmn 😁👌

File: src/components/Home.js (hunk 1 above)
Comment (keyy123) [reply src/components/Home.js:138]: I will revert back changes in Home.js then 🙂.

**Comment by keyy123** [comment]
@RubenSmn what component had that class?

**Comment by keyy123** [comment]
@RubenSmn, MakeStyles has a thing with it classes. In dev environments, class name is "makestyles-${className}-${#}" (i.e., classes.link === makestyles-link-1) but in production, the classname is "jss${#}" (classes.link === jss1). In this case, every `<Link>` component in the app (~20 in web-map-client) will have a jss1 class name in prod b/c we attached make styles link class to the sub-component within the link component. 

TLDR: The styles that you see come from MakeStyles  - it's called jss1 in prod and makestyles-link-1 in dev

**Comment by RubenSmn** [comment]
So there is not really a problem with "not being able to set class styles". The code is working as expected, when you add the `display: 'block'` to the `makeStyles` it will be applied to the `a` tag in both the **dev** and **prod** environment.

@keyy123 about the link colors I think it is fine to leave it like this. Our links all have different colors, so if we want to change it we'll probably do it by overriding the style.

**Review by RubenSmn — COMMENTED** [review]
If you could move the **inline** style to the **class style** then I think the issue will be solved

File: src/components/Home.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { Brightness1 } from '@mui/icons-material';
```
Comment (RubenSmn) [inline src/components/Home.js:1]: Could you remove this unused import

File: src/components/Link.js [hunk 5]
```diff
@@ -1,16 +1,22 @@
 /* eslint-disable jsx-a11y/anchor-has-content */
 /* eslint-disable react/display-name */
+import { Block } from '@mui/icons-material';
```
Comment (RubenSmn) [inline src/components/Link.js:3]: Same unused here

**Review by RubenSmn — CHANGES_REQUESTED** [review]

**Review by RubenSmn — APPROVED** [review]
Awesome!

**Comment by RubenSmn** [comment]
@keyy123 Could you resolve the merge conflict?

**Comment by keyy123** [comment]
@RubenSmn 😁👌

**Comment by dadiorchen** [comment]
Because we discard the issue #830 so let's close this PR for now.
